### PR TITLE
adapt to improved type hints and fix edge case issues with `Package-files`

### PR DESCRIPTION
**Pull request by radoering**
# Pull Request Check List

Requires: python-poetry/poetry-core#904

Fix edge case issues (and corrupt tests) uncovered by python-poetry/poetry-core#904.

## Summary by Sourcery

Align package file handling with updated typing and fix edge cases around missing or invalid file hashes.

Bug Fixes:
- Correct package file key usage in chooser, executor, and locker logic to prevent mismatches and corrupted tests.
- Skip files without resolvable hashes in HTTP repository link conversion to avoid writing invalid lockfile entries.

Enhancements:
- Adopt the shared PackageFile type in repositories, inspection, executor tests, and chooser tests for stricter type safety and consistency.

Build:
- Update poetry-core dependency to a specific Git revision that introduces improved package file typing.

Tests:
- Adjust installation chooser and executor tests to use the new PackageFile structure and align with updated hash handling.

### CODE REVIEW

File: tests/installation/test_executor.py [hunk 1]
```diff
@@ -1819,7 +1820,7 @@ def test_other_error(
     ],
 )
 def test_executor_known_hashes(
-    package_files: list[dict[str, str]],
+    package_files: list[PackageFile],
```
[the remark this answers is not in the document]
Comment (radoering) [reply tests/installation/test_executor.py:1823]: probably not worth the effort

File: tests/installation/test_chooser.py [hunk 2]
```diff
@@ -211,7 +213,7 @@ def test_chooser_chooses_distributions_that_match_the_package_hashes(
     chooser = Chooser(pool, env)
 
     package = Package("isort", "4.3.4")
-    files = [
+    files: list[PackageFile] = [
```
[the remark this answers is not in the document]
Comment (radoering) [reply tests/installation/test_chooser.py:216]: Not worth the effort because I am not sure if this is even possible or only a theoretical case.
